lib/util.js

```diff
@@ -63,9 +64,57 @@ function sortLibraries(a, b) {
   return a.type.localeCompare(b.type);
 }
 
+// spinner constant
```

Comment:
let's avoid module level state, let's assume the algorithms in question can be re-run.

Reply:
Moved this whole logic into a class so that we avoid the module level mutable state.